src/bregman.jl

```diff
@@ -124,7 +126,7 @@ function bregman(funobj::Function, x::AbstractArray{T}, options::BregmanParams,
         # Update z variable
         @. z = z + d
         # Get λ at first iteration
-        i == 1 && (λ = abs(T(quantile(abs.(z), options.quantile))))
+        i == 1 && (sol.λ = λ = isnothing(options.lambda) ? abs(T(quantile(abs.(z), options.quantile))) : abs(T(options.lambda)))
```

Comment:
That's a lot of if in one line but I guess.

I'm still waiting on an answer as to why you would ever end up with a complex-valued `x`

Reply:
So that will be easy if I want to solve `AC'x=b` where x is the complex curvelet coefficients. It is easier for me to take this form if I want to do weightings on `x` later

Reply:
This is not the right way to use this as you are not solving the correct problem. You are always solving the `L1-l2` elastic net on the curvelet coefficient so `x` is your image and is real. if you give x as the input and C' instead of C you are solving sparsity in the image domain.

If you want to do weighting you need to provide `W*c`

Reply:
OK I see your point. However, I think they are different ways to form the optimization problem. Either way works because they should be mathematically equivalent to solve for either the image or its curvelet coefficients. So I still think a PR to fix the support of complex number is needed. Any thought?

Reply:
Either way does not work. You are confusing mathematical equivalence and implementation equivalence you can't make a program magically understand a change of variable (which is what you are doing). The documentation fo this implementation is very explicit about what it solves.

I don't have an issue with complex number supports but you are not doing what you think you are doing and it has to be for the case where `x` is complex, the  actual primal `x`.

Reply:
1. solving Ax = b where x is primal image, z is dual curvelet coefficient. Soft-thresholding is done by forward curvelet, threshold and adjoint curvelet.
2. sovling AC'x=b where x is primal curvelet coefficient, z is dual curvelet coefficient. Soft-thresholding is done w/o curvelet, just threshold.

Why aren't they equivalent?

Reply:
You are not listening nor reading what I said

Reply:
The problem of `provide W*C instead of C` is that its adjoint is not inverse. Then during soft-thresholding we have to do a W inverse.

Reply:
That's because you are trying to do a weighted l1 which is a different problem. You could add weighted l1 with the standard identity as default but again you need to be careful about what problem you are solving.

Reply:
OK I agree what you are saying now but we don't have to add anything new: just solve `AC'Wx = b` and use the identity as the sparsity transform will do the work. The solution is `C'WX` in the end. That's much simpler than adding a weighted l1

Reply:
That's not the same as you will computethe threshold on `W*z` not on `z`.

Reply:
No we don't have to. The forward and adjoint of `W` can already bump the entries on support 2 times so these entries are much easier to pass the threshold